### PR TITLE
fix(helm): set cloudsql-proxy as sidecar container to allow initializer and dbmigration to run

### PR DESCRIPTION
**Pull request by jndeverteuil**
## :warning: Note on feature completeness :warning:

We are narrowing the scope of acceptable enhancements to DefectDojo in preparation for v3. Learn more here:
https://github.com/DefectDojo/django-DefectDojo/blob/master/readme-docs/CONTRIBUTING.md

**Description**

This PR is to fix an issue when using `cloudsql-proxy` as per the following discussion: https://github.com/DefectDojo/django-DefectDojo/issues/7235

> When using Cloud SQL as a managed database, the Job related to Initializer isn't completed because your Sidecar is still running. Sidecar is cloudsql-proxy.

By setting the `cloudsql-proxy` as a special `initContainer` of type`sidecar` (see [relevant kubernetes doc](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/)), we allow the `wait-for-db` initContainer to reach the database.

> Kubernetes implements sidecar containers as a special case of [init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/); sidecar containers remain running after Pod startup.

> These restartable sidecar containers are independent from other init containers and from the main application container(s) within the same pod.

> If an init container is created with its restartPolicy set to Always, it will start and remain running during the entire life of the Pod.

> If you define a Job that uses sidecar using Kubernetes-style init containers, the sidecar container in each Pod does not prevent the Job from completing after the main container has finished.

image

image

Without this, `wait-for-db` is never allowed to reach the database since `cloudsql-proxy` is set as a regular `container` which are running only after `initcontainer` are finished.

### CODE REVIEW

File: helm/defectdojo/templates/celery-beat-deployment.yaml
Comment (tidusete) [inline helm/defectdojo/templates/celery-beat-deployment.yaml:69]: I know that the dbMigrationChecker is enabled by default, but shouldn't be outside this conditional? I think using CloudSQL is not related with the fact that you must have enabled dbMigrationChecker
I tested your MR using an external CloudSQL instance and it only works if dbMigrationChecker is enabled.

File: helm/defectdojo/templates/celery-worker-deployment.yaml
Comment (tidusete) [inline helm/defectdojo/templates/celery-worker-deployment.yaml:67]: Same problem here

File: helm/defectdojo/templates/django-deployment.yaml
Comment (tidusete) [inline helm/defectdojo/templates/django-deployment.yaml:85]: Same problem here

File: helm/defectdojo/templates/celery-beat-deployment.yaml
Comment (kiblik) [reply helm/defectdojo/templates/celery-beat-deployment.yaml:69]: I agree with out in this point

File: helm/defectdojo/templates/django-deployment.yaml
Comment (kiblik) [inline helm/defectdojo/templates/django-deployment.yaml:85]:
```suggestion
      {{- if or .Values.dbMigrationChecker.enabled .Values.cloudsql.enabled }}
```

File: helm/defectdojo/templates/django-deployment.yaml
Comment (kiblik) [inline helm/defectdojo/templates/django-deployment.yaml:110]:
```suggestion
      {{- end }}
      {{- end }}
```

**Review by kiblik — CHANGES_REQUESTED** [review]
I suppose this should help.

File: helm/defectdojo/templates/celery-beat-deployment.yaml
Comment (tidusete) [reply helm/defectdojo/templates/celery-beat-deployment.yaml]: Exactly

**Review by tidusete — COMMENTED** [review]
I agree on all what @kiblik suggest

File: helm/defectdojo/templates/django-deployment.yaml
Comment (jndeverteuil) [reply helm/defectdojo/templates/django-deployment.yaml]: Nice we had the same conclusion, I just hadn't seen the suggestion here otherwise I would've implemented it instead of creating my commit, but same exact line used

File: helm/defectdojo/templates/celery-beat-deployment.yaml
Comment (jndeverteuil) [reply helm/defectdojo/templates/celery-beat-deployment.yaml]: Thanks for the feedback, I solved it in the new commit

Comment (jndeverteuil) [reply helm/defectdojo/templates/celery-beat-deployment.yaml]: Thank you, implemented!

**Review by kiblik — APPROVED** [review]

File: helm/defectdojo/Chart.yaml
Comment (Maffooch) [inline helm/defectdojo/Chart.yaml:5]:
```suggestion
version: 1.6.155-dev
```
This only gets revved on release

**Review by tidusete — APPROVED** [review]
lgtm

**Comment by tidusete** [comment]
Is there any issue on it? I'm waiting for this fix to end up on a release :pray:

**Review by mtesauro — APPROVED** [review]
Approved

**Comment by tidusete** [comment]
Is there any problem with the MR? It looks like it's still Open

**Comment by mtesauro** [comment]
@tidusete 
> Is there any problem with the MR? It looks like it's still Open

PRs require 4 approvals from DefectDojo maintainers and this one is 2 short currently.

**Review by Maffooch — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Pull request merged**
